contracts/stake/src/tests/distribution.rs

```diff
@@ -976,8 +1069,86 @@ fn panic_when_adding_same_distribution_twice() {
     let lp_token = deploy_token_contract(&env, &admin);
     let reward_token = deploy_token_contract(&env, &admin);
 
-    let staking = deploy_staking_contract(&env, admin.clone(), &lp_token.address, &manager, &owner);
+    let staking = deploy_staking_contract(
+        &env,
+        admin.clone(),
+        &lp_token.address,
+        &manager,
+        &owner,
+        &50u32,
+    );
+
+    staking.create_distribution_flow(&manager, &reward_token.address);
+    staking.create_distribution_flow(&manager, &reward_token.address);
+}
+
+#[should_panic(expected = "Stake: Fund distribution: Curve complexity validation failed")]
+#[test]
+fn panic_when_funding_distribution_above_end_distribution_timestamp() {
+    const DISTRIBUTION_MAX_COMPLEXITY: u32 = 1;
+    let env = Env::default();
+    env.mock_all_auths();
+
+    let admin = Address::generate(&env);
+    let manager = Address::generate(&env);
+    let owner = Address::generate(&env);
+    let lp_token = deploy_token_contract(&env, &admin);
+    let reward_token = deploy_token_contract(&env, &admin);
+
+    let staking = deploy_staking_contract(
+        &env,
+        admin.clone(),
+        &lp_token.address,
+        &manager,
+        &owner,
+        &DISTRIBUTION_MAX_COMPLEXITY,
+    );
 
     staking.create_distribution_flow(&manager, &reward_token.address);
+
+    reward_token.mint(&admin, &2000);
+
+    staking.fund_distribution(&admin, &0, &600, &reward_token.address, &1000);
+    staking.fund_distribution(&admin, &0, &600, &reward_token.address, &1000);
+
+    env.ledger().with_mut(|li| {
+        li.timestamp = 601;
+    });
+}
+
+#[test]
+#[ignore = "cannot access the state directly to check the reward curve"]
+fn test_check_complexity_over_time() {
+    let env = Env::default();
+    env.mock_all_auths();
+
+    let admin = Address::generate(&env);
+    let contract_addr = Address::generate(&env);
+    let manager = Address::generate(&env);
+    let owner = Address::generate(&env);
+    let lp_token = deploy_token_contract(&env, &admin);
+    let reward_token = deploy_token_contract(&env, &admin);
+
+    let staking = StakingClient::new(
+        &env,
+        &env.register_contract(Some(&contract_addr), Staking {}),
+    );
+
+    staking.initialize(
+        &admin,
+        &lp_token.address,
+        &1_000,
+        &1_000,
+        &manager,
+        &owner,
+        &10,
+    );
+
+    // dbg!(env.current_contract_address());
     staking.create_distribution_flow(&manager, &reward_token.address);
+
+    reward_token.mint(&admin, &2000);
+
+    staking.fund_distribution(&admin, &0, &600, &reward_token.address, &1000);
+    let _result = get_reward_curve(&env, &reward_token.address);
```

Comment:
Just remove this test since it doesn't make sense in a first place. It's more of a unit test anyway.